paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -61,414 +61,428 @@ static mkldnn::memory::data_type GetDstType(bool is_int8, bool is_bfloat16,
 
-      this->AcquireBackwardWeightsPrimitiveDescriptor(
-          mkldnn::algorithm::convolution_direct, src_md, diff_weights_md,
-          diff_dst_md, strides, dilations_dims, mkldnn_paddings[0],
-          mkldnn_paddings[1]);
-    }
-  }
+    std::vector<int> strides_temp = ctx.Attr<std::vector<int>>("strides");
+    std::vector<int64_t> strides(begin(strides_temp), end(strides_temp));
 
-  std::shared_ptr<std::tuple<float, std::vector<float>>> get_int8_bias_scales(
-      const framework::ExecutionContext& ctx) {
-    // Get scales int8 bias key
-    const std::string key_bs = this->key_ + "@bs";
+    std::vector<int> paddings_temp = ctx.Attr<std::vector<int>>("paddings");
+    std::vector<int64_t> paddings(begin(paddings_temp), end(paddings_temp));
 
-    // Scales for int8 bias are to be cached to avoid
-    // computing them each iteration
-    auto bias_scale_tuple =
-        std::static_pointer_cast<std::tuple<float, std::vector<float>>>(
-            this->dev_ctx_.GetBlob(key_bs));
-    if (bias_scale_tuple) return bias_scale_tuple;
+    std::vector<int> dilations_temp = ctx.Attr<std::vector<int>>("dilations");
+    std::vector<int64_t> dilations(begin(dilations_temp), end(dilations_temp));
 
-    const auto* filter = ctx.Input<Tensor>("Filter");
-    const auto& weights_tz = framework::vectorize(filter->dims());
-    const int groups = std::max(ctx.Attr<int>("groups"), 1);
+    auto input_dims = in->dims();
+    auto data_dims = framework::slice_ddim(input_dims, 2, input_dims.size());
+    auto filter_dims = filter->dims();
+    auto filter_data_dims =
+        framework::slice_ddim(filter_dims, 2, filter_dims.size());
+    auto ksize = framework::vectorize(filter_data_dims);
 
-    const auto& scale_weights_data =
-        ctx.Attr<std::vector<float>>("Scale_weights");
-    const auto& scale_in_data = ctx.Attr<float>("Scale_in");
+    std::string padding_algorithm = ctx.Attr<std::string>("padding_algorithm");
+    UpdatePaddingAndDilation(&paddings, &dilations, padding_algorithm,
+                             data_dims, strides, ksize);
 
-    bool is_multi_channel = scale_weights_data.size() > 1;
-    int mask_reorder = is_multi_channel ? 1 << 0 : 1;
+    auto src_tz = framework::vectorize(in->dims());
+    auto weights_tz = framework::vectorize(filter->dims());
 
-    int count = 1;
-    if (is_multi_channel) {
-      count *= weights_tz[0];
-      if (groups > 1) {
-        count *= weights_tz[1];
+    int groups = ctx.Attr<int>("groups");
+    int g = std::max(groups, 1);
+    platform::GetGroupConvWeightsTz(weights_tz, g);
+    auto dst_tz = framework::vectorize(out_grad->dims());
+
+    /* create memory descriptor for conv backward without specified format
+     * ('any') which lets a primitive (conv backward in this case) choose
+     * the memory format preferred for best performance
+     */
+    const auto chosen_memory_format = MKLDNNMemoryFormat::any;
+    const auto weights_format = MKLDNNMemoryFormat::any;
```

Comment:
What is the point of having two identical variables here? Every format im conv should use any, do I think that this redundancy is not necessary

Reply:
some legacy remaining. I